form.js

```diff
@@ -96,13 +99,50 @@ function node_type_change_hander() {
   toggle_cuda_version_visibility();
 }
 
+/**
+ * Sets the change handler for the cluster
+ */
+function set_cluster_change_handler(){
+  let cluster_input = $('#batch_connect_session_context_cluster');
+  cluster_input.change(cluster_change_handler);
+}
+
+/**
+ * Update UI when cluster changes; node_type_change_handler must be called again for each cluster
+ */
+function cluster_change_handler(){
+  toggle_node_type_visibility();
+  set_node_type_change_handler();
```

Comment:
You only need to set the change handler once.

Reply:
The node type id changes to `batch_connect_session_context_owens_node_type` and pitzer_node_type. It has to be called for each cluster.

Reply:
Right, and I guess that ties into the other comment, that there can be only one `node_type` form item. Toggling an event handler's source event is not straight forward. I mean "this handler looks for this element to change" is simple to reason about.